File: dotnet/01-commands/doc/Banking_Kata/AccountStatementPrint.cs
namespace Banking_Kata
{
    internal static class AccountStatementPrint
    {
        public static string PrintStatement(List<BankingTransactionDTO> transactiontList)
        {
            // Initialize the printed statement with the header
            string printedStatement = "Date Amount Balance\n";

            // Iterate over each transaction in the list
            foreach (BankingTransactionDTO transaction in transactiontList)
            {
                // Format each transaction's details and append to the printed statement
                printedStatement += $"{transaction.date.ToString("d")} " +
                    $"{(transaction.amount >= 0 ? "+" : "")}{transaction.amount} " +
                    $"{transaction.balance}\n";
            }

            // Return the final formatted statement
            return printedStatement;
        }

    }
}

File: cpp/01-commands/doc/Banking_Kata/AccountStatementPrint.cpp
#include "pch.h"
#include "AccountStatementPrint.h"

using namespace std::chrono;

std::string PrintDate(const timepoint& date)
{
    year_month_day ymd(floor<days>(date));
    return std::format("{:04}-{:02}-{:02}", 
               static_cast<int>(ymd.year()), 
               static_cast<unsigned>(ymd.month()), 
               static_cast<unsigned>(ymd.day()));
}


std::string PrintStatement(const std::vector<BankingTransactionDTO>& statement)
{
    std::string printedStatement = "Date Amount Balance\n";
    
    for (const auto& transaction : statement)
    {
        printedStatement += std::format("{} {:+} {}\n",
            PrintDate(transaction.getDate()),
            transaction.getAmount(),
            transaction.getBalance());
    }

    return printedStatement;
}
